internal/provider/resource_apply_module.go

```diff
@@ -260,8 +317,13 @@ func (t *applyModuleResource) Read(ctx context.Context,
 		state.ModuleVersion = types.StringNull()
 	}
 
-	// TODO: Eventually, when the API and SDK support speculative plan runs with a module source,
-	// this should do a speculative plan run here to determine whether changes are needed.
+	// Add namespace paths to the variables.
+	outVars, diags := t.addNamespacePaths(ctx, &state.Variables, state.WorkspacePath.ValueString())
```

Comment:
This `addNamespacePaths` call should not be needed. The namespace path will not be the same for all resolved variables. The SDK has a function to get the Run variables which includes the namespace path for each variable.

Reply:
I searched for an SDK function that would return the resolved run variables after a run had finished.  I was not able to find it.  I'll search again.

Reply:
I found GetRunVariables.  However, the doRun function does not return the required real/Tharsis ID of the run.  I can either add the run's real/Tharsis metadata ID to the doRunOutput type and get the variables outside of doRun; or I can get the variable inside doRun and add the list of variables to the doRunOutput type.  I the the latter would be preferable.